state-chain/pallets/cf-elections/src/electoral_systems/mocks.rs

```diff
@@ -79,36 +83,48 @@ where
 	// Useful for testing check_consensus since we already have an election.
 	pub fn build_with_initial_election(self) -> TestContext<ES> {
 		let setup = self.clone();
-		let mut electoral_access = MockAccess::<ES>::new(
-			self.unsynchronised_state,
-			self.unsynchronised_settings,
-			self.electoral_settings,
-		);
+
+		// We need to clear the storage at every build so if there are multiple test contexts used
+		// within a single test they do not conflict.
+		MockStorageAccess::clear_storage();
 
 		let (election_identifier_extra, election_properties, election_state) =
 			self.initial_election_state.unwrap_or_default();
 
-		let election = electoral_access
-			.new_election(election_identifier_extra, election_properties, election_state)
-			.unwrap();
+		// These are synchronised with respect to an election, so for simplicity in the tests
+		// we just assign them to an election.
+		MockStorageAccess::set_electoral_settings::<ES>(setup.electoral_settings.clone());
+		MockStorageAccess::set_unsynchronised_state::<ES>(setup.unsynchronised_state.clone());
+		MockStorageAccess::set_unsynchronised_settings::<ES>(setup.unsynchronised_settings.clone());
```

Comment:
I don't really understand the comment here. How are these 'assigned to an election'?

Reply:
Made comment more explicit.